src/clip.js

```diff
@@ -41,13 +41,17 @@ function clip(features, scale, k1, k2, axis, intersect, minAll, maxAll) {
         if (slices.length) {
             // if a feature got clipped, it will likely get clipped on the next zoom level as well,
             // so there's no need to recalculate bboxes
-            clipped.push({
+            var clippedFeature = {
                 geometry: slices,
                 type: type,
                 tags: features[i].tags || null,
                 min: feature.min,
                 max: feature.max
-            });
+            };
+            if (typeof feature.id !== 'undefined') {
+                clippedFeature.id = feature.id;
+            }
```

Comment:
Are you serializing the tiles on disk, or are you concerned about memory footprint? The overhead shouldn't be significant. `delete` is not an option because it harms performance significantly. Also, I'd prefer `id: null` for non-existing ids rather than `id: undefined`.